### ISSUE TITLE
progressive encoding of lossless images is broken

### ISSUE THREAD

**Issue by Sur3** [libjxl/libjxl#2812]
Hi I tried to compress a lossless image progressivly, but the result is more than twice the file size of the lossless version without progressive:
```
21885191 Red_Fuji_southern_wind_clear_morning.png
11719861 Red_Fuji_southern_wind_clear_morning.jxl
26925947 Red_Fuji_southern_wind_clear_morning-progressive.jxl

```
I was using the latest main branch commit of libjxl but earlier versions seem to have the same problem.

When using quality 85 instead of lossless this seems to work fine but version 0.8.1 of libjxl produced smaller images at the same quality setting compared to the main branch; is this intentional?
```
2386357 Red_Fuji_southern_wind_clear_morning-q85-main.jxl
2274502 Red_Fuji_southern_wind_clear_morning-q85-0.8.1.jxl
2545784 Red_Fuji_southern_wind_clear_morning-q85-progressive-main.jxl
2392863 Red_Fuji_southern_wind_clear_morning-q85-progressive-0.8.1.jxl
```

**Comment by Sur3** [comment]
Probably related:

I used main branch from yesterday to encode images with 0.00 lossless distance and 0.01 lossy distance and found the lossless results to be smaller in size:

```
18701045 Red_Fuji_southern_wind_clear_morning-distance001.jxl
11719861 Red_Fuji_southern_wind_clear_morning-distance000.jxl
```

**Comment by mo271** [comment]
Non-intuitively, it can happen that lossless compression will result in a smaller files than lossy compression, especially for images that are not quite photographic content.

**Comment by Sur3** [comment]
> Non-intuitively, it can happen that lossless compression will result in a smaller files than lossy compression, especially for images that are not quite photographic content.

Yeah thats not the major problem here, but that the progressive lossless image is more than twice as large as the non-progressive lossless image.

**Comment by sergeevabc** [comment]
Windows 7 x64, cjxl 0.8.2, [image.jpg](https://s7.krakenfiles.com/uploads/09-11-2023/OSMbbh2YVA/image.jpg)

* _cjxl image.jpg output.jxl_ works fine
  ```
  JPEG XL encoder v0.8.2 954b460 [SSSE3,Unknown]
  Note: Implicit-default for JPEG is lossless-transcoding.
  Read JPEG image with 4244645 bytes.
  Encoding [Container | JPEG, lossless transcode, effort: 7 | JPEG reconstruction data],
  Compressed to 3466260 bytes including container
  ```
* _cjxl image.jpg output.jxl -p_ throws an error as follows
  ```
  ...
  D:\a\libjxl\libjxl\lib\jxl\enc_frame.cc:908: JXL_CHECK: enc_state_->passes.size() == 1
  ```
  ![2023-11-09 cjxl 082 progressive](https://github.com/libjxl/libjxl/assets/373718/cd674c70-5c73-4c27-a55d-99eb0d418cc1)

  

Why?

**Comment by sergeevabc** [comment]
Err... Hello?

**Comment by kleisauke** [comment]
> * _cjxl image.jpg output.jxl -p_ throws an error as follows
>   ```
>   ...
>   D:\a\libjxl\libjxl\lib\jxl\enc_frame.cc:908: JXL_CHECK: enc_state_->passes.size() == 1
>   ```

This is being tracked in issue #1988.

**Comment by jonnyawsom3** [comment]
Effort 2, 3 and 4 seem to work correctly. Effort 5+ have a severe filesize increase. Disabling local pallete seems to help, but still isn't close to effort 4.
